react/components/atoms/container-swipeable-horizontal/container-swipeable-horizontal.js

```diff
@@ -0,0 +1,284 @@
+        const distance = Math.abs(offset);
+        return distance > 0 && distance > this.slowDownThreshold;
+    }
+
+    setSwipingDirection(offset) {
+        if (offset > 0 && this.props.swipeLeftEnabled) {
+            this.setState({ swipingDirection: "left" });
+            return true;
+        } else if (offset < 0 && this.props.swipeRightEnabled) {
+            this.setState({ swipingDirection: "right" });
+            return true;
+        }
+
+        this.setState({ swipingDirection: undefined });
+        return false;
+    }
+
+    onMoveShouldSetPanResponderCapture = (event, gestureState) => {
+        if (!this.props.swipeLeftEnabled && !this.props.swipeRightEnabled) {
+            return false;
+        }
+        return !this.animating;
+    };
+
+    onPanResponderMove = (event, gestureState) => {
+        if (this.animating) return;
+
+        if (!this.setSwipingDirection(gestureState.dx)) {
+            this.state.animationPositionX.setValue(0);
+            return;
+        }
+
+        if (this.isAfterThreshold(gestureState.dx)) {
+            const sign = this.state.swipingDirection === "left" ? 1 : -1;
+            const base = sign * this.slowDownThreshold;
+            const offset = base + (gestureState.dx - base) / this.props.afterThresholdSlowdown;
+
+            this.state.animationPositionX.setValue(offset);
+        } else {
+            this.state.animationPositionX.setValue(gestureState.dx);
+        }
+    };
+
+    onPanResponderRelease = (eventt, gestureState) => {
+        this.animating = true;
+
+        if (this.isAfterThreshold(gestureState.dx)) {
+            if (this.state.swipingDirection === "left") {
+                this.props.onLeftOptionTrigger();
+            } else {
+                this.props.onRightOptionTrigger();
+            }
+        }
+
+        Animated.timing(this.state.animationPositionX, {
+            toValue: 0
+        }).start(() => {
+            this.animating = false;
+        });
+    };
+
+    _leftOptionTextStyle() {
+        return [
+            styles.textOption,
+            { color: this.props.leftOptionTextColor },
+            this.props.leftOptionTextStyle
+        ];
+    }
+
+    _rightOptionTextStyle() {
+        return [
+            styles.textOption,
+            { color: this.props.rightOptionTextColor },
+            this.props.rightOptionTextStyle
+        ];
+    }
+
+    _rightOptionsGradientStyle() {
+        return [styles.containerGradient, styles.containerGradientOptionRight];
+    }
+
+    _contentStyle = () => {
+        return [{ transform: [{ translateX: this.state.animationPositionX }] }];
+    };
+
+    _style = () => {
+        return [styles.containerSwipeableHorizontal, this.props.style];
+    };
+
+    _renderLeftOption = () => {
+        return (
+            <LinearGradient
+                start={{ x: 0, y: 0 }}
+                end={{ x: 1, y: 0 }}
+                colors={this.props.leftOptionGradientColors}

```

Comment:
@FlaviooLima this needs to be configurable in the same way as our `Button`'s gradient is.